### DISCUSSION TITLE
Convert "flat" chiral center to tetrahedral

### DISCUSSION THREAD

**Discussion by spparel** [rdkit/rdkit#6962] [Q&A]
Hi all,
Historically, we have stored racemic mixture of enantiomers with one chiral center by drawing this chiral center "flat" (which is not ideal, considering that "flat" means we don't know anything about a given structure). We would like to revise this approach by applying enhanced stereochemistry rules. This would be both for rendering as well as redrawing historical structures in our corporate database.
We are able to identify such cases by using findMolChiralCenters, for instance, and we are also able to assign an "and1" atom label. 

Is there a straightforward way to convert the flat chiral center to a properly drawn chiral center with a wedge bond ? I.e. we want to achieve this:
![image](https://github.com/rdkit/rdkit/assets/135133375/ba9361bd-aef0-4a7f-941a-4a605b915cba)
Thank you for your help

**Comment by bp-kelley** [comment]
This function should do what you want, I'm planning on putting it into the RDKit fairly soon.  The secret is in:

Chem.CreateStereoGroup(

```
from rdkit import Chem

def determineStereoGroups(mol, determine_unassigned_stereo=True):
  """This converts unassigned stereo to stereo groups

   Args:
   :mol: Molecule to convert
   :determine_unassigned_stereo: if True generate stereo groups from unassigned stereo [True]

  Returns:
   New molecule with assigned stereo groups
  """
  rwmol = Chem.RWMol(mol)
  Chem.FindPotentialStereoBonds(rwmol)
  
  min_idx = 0
  atoms_in_stereo_groups = set()
  for sg in rwmol.GetStereoGroups():
    for atom in sg.GetAtoms():
      atoms_in_stereo_groups.add(atom.GetIdx())
                                
  sgs = list(rwmol.GetStereoGroups())

  new_sgs = []
  for atom in rwmol.GetAtoms():
    if atom.HasProp("_ChiralityPossible"):
      tag = atom.GetChiralTag()

      if tag == Chem.ChiralType.CHI_UNSPECIFIED:
        if determine_unassigned_stereo:
          atom.SetChiralTag(Chem.ChiralType.CHI_TETRAHEDRAL_CW)
          new_sgs.append(Chem.CreateStereoGroup(Chem.StereoGroupType.STEREO_AND,rwmol,[atom.GetIdx()]))

      elif tag in [Chem.ChiralType.CHI_TETRAHEDRAL_CCW, Chem.ChiralType.CHI_TETRAHEDRAL_CW]:
        if atom.GetIdx() not in atoms_in_stereo_groups:
          new_sgs.append(Chem.CreateStereoGroup(Chem.StereoGroupType.STEREO_ABSOLUTE,rwmol,[atom.GetIdx()]))

  if new_sgs:
    rwmol.SetStereoGroups(sgs + new_sgs)

  return rwmol
```

```
m = Chem.MolFromSmiles("NC(CC1=CC=CC=C1)C(O)=O")
m = determineStereoGroups(m)

print(Chem.MolToCXSmiles(m))
```


```
N[C@H](Cc1ccccc1)C(=O)O |&1:1|
```

**Comment by spparel** [comment]
Hi @bp-kelley,

This is very useful and exactly what I was looking for. We certainly can build on this to implement our required logic to process legacy information in our database.

Thank you !
